teps/0085-per-namespace-controller-configuration.md

```diff
@@ -79,6 +80,42 @@ As a contributor, I need to test my behavioral changes to ensure that they work
 - Operator can allow for configuration to be defined on per-namespace basis
 - User can specify and use a customized configuration for a given namespace
 
+## Proposal
+
+To enable the `tekton-pipelines-controller`'s feature flags configuration per namespace, we propose adding a new `configmap` with the same name `feature-flags` in each of the candidate namespaces, which contains the properties to override for all the `Pipelineruns` created in this specific namespace.
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: feature-flags
+data:
+  disable-home-env-overwrite: "false"
+  scope-when-expressions-to-task: "true"
+```
+
+The candidate namespaces are included in a new environment variable `CONFIG_FEATURE_FLAGS_CUSTOMIZATION_NAMESPACES`.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: tekton-pipelines-controller
+  namespace: tekton-pipelines
+spec:
+  template:
+    spec:
+      containers:
+      - name: tekton-pipelines-controller
+        image: gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/controller
+        env:
+        - name: CONFIG_FEATURE_FLAGS_CUSTOMIZATION_NAMESPACES
```

Comment:
is it a conscious decision to use a env variable instead of an entry in the tekton-pipelines feature-flags configmap?

Reply:
One of the nice benefits of making this a field in the feature-flags configmap would be that adding new namespaces wouldn't require a controller restart.

Reply:
- Is this environment variable required or optional ? a.k.a. what happens when it's not present ? 
- Do we see value in supporting regular expression here ?

Asking those question as I think one of the main use-case here might be to enable this for all namespace but a certain set of them for example. It wouldn't scale if each time a new namespace is created in the cluster, the cluster-admin need to updated the pipeline deployment spec.